src/main/flight/rpm_filter.c

```diff
@@ -52,57 +52,63 @@
 
 static pt1Filter_t rpmFilters[MAX_SUPPORTED_MOTORS];
 
-typedef struct rpmNotchFilter_s
-{
-    uint8_t harmonics;
-    float   minHz;
-    float   maxHz;
-    float   q;
-    float   loopTime;
+typedef struct rpmNotchFilter_s {
+
+    uint8_t  harmonics;
+    uint8_t  minHz;
+    uint16_t maxHz;
+    uint16_t fadeRangeHz;
+    float    q;
+    uint32_t looptime;
 
     biquadFilter_t notch[XYZ_AXIS_COUNT][MAX_SUPPORTED_MOTORS][RPM_FILTER_MAXHARMONICS];
+
 } rpmNotchFilter_t;
 
 FAST_DATA_ZERO_INIT static float   erpmToHz;
 FAST_DATA_ZERO_INIT static float   filteredMotorErpm[MAX_SUPPORTED_MOTORS];
+FAST_DATA_ZERO_INIT static float   motorFrequency[MAX_SUPPORTED_MOTORS];
 FAST_DATA_ZERO_INIT static float   minMotorFrequency;
 FAST_DATA_ZERO_INIT static uint8_t numberFilters;
 FAST_DATA_ZERO_INIT static uint8_t numberRpmNotchFilters;
 FAST_DATA_ZERO_INIT static uint8_t filterUpdatesPerIteration;
 FAST_DATA_ZERO_INIT static float   pidLooptime;
 FAST_DATA_ZERO_INIT static rpmNotchFilter_t filters[2];
-FAST_DATA_ZERO_INIT static rpmNotchFilter_t* gyroFilter;
+FAST_DATA_ZERO_INIT static rpmNotchFilter_t *gyroFilter;
 
 FAST_DATA_ZERO_INIT static uint8_t currentMotor;
 FAST_DATA_ZERO_INIT static uint8_t currentHarmonic;
 FAST_DATA_ZERO_INIT static uint8_t currentFilterNumber;
-FAST_DATA static rpmNotchFilter_t* currentFilter = &filters[0];
+FAST_DATA static rpmNotchFilter_t *currentFilter = &filters[0];
 
 
 
-PG_REGISTER_WITH_RESET_FN(rpmFilterConfig_t, rpmFilterConfig, PG_RPM_FILTER_CONFIG, 4);
+PG_REGISTER_WITH_RESET_FN(rpmFilterConfig_t, rpmFilterConfig, PG_RPM_FILTER_CONFIG, 5);
 
 void pgResetFn_rpmFilterConfig(rpmFilterConfig_t *config)
 {
     config->gyro_rpm_notch_harmonics = 3;
     config->gyro_rpm_notch_min = 100;
+    config->gyro_rpm_notch_fade_range_hz = 100;
     config->gyro_rpm_notch_q = 500;
 
     config->rpm_lpf = 150;
 }
 
-static void rpmNotchFilterInit(rpmNotchFilter_t* filter, int harmonics, int minHz, int q, float looptime)
+static void rpmNotchFilterInit(rpmNotchFilter_t *filter, const rpmFilterConfig_t *config, const uint32_t looptime)
 {
-    filter->harmonics = harmonics;
-    filter->minHz = minHz;
-    filter->q = q / 100.0f;
-    filter->loopTime = looptime;
-
-    for (int axis = 0; axis < XYZ_AXIS_COUNT; axis++) {
-        for (int motor = 0; motor < getMotorCount(); motor++) {
-            for (int i = 0; i < harmonics; i++) {
+    filter->harmonics = config->gyro_rpm_notch_harmonics;
+    filter->minHz = config->gyro_rpm_notch_min;
+    filter->maxHz = 0.48f * 1e6f / looptime; // don't go quite to nyquist to avoid oscillations
+    filter->fadeRangeHz = config->gyro_rpm_notch_fade_range_hz;
+    filter->q = config->gyro_rpm_notch_q / 100.0f;
+    filter->looptime = looptime;
+
+    for (uint8_t axis = 0; axis < XYZ_AXIS_COUNT; axis++) {
```

Comment:
Using `unsigned` here will potentially be faster, as it ensures that the compiler does not insert range checks.

Reply:
Good point, will change it back to `int`